packages/pigeon/platform_tests/test_plugin/linux/test_plugin.cc

```diff
@@ -0,0 +1,1854 @@
+    .echo_async_nullable_bool = echo_async_nullable_bool,
+    .echo_async_nullable_string = echo_async_nullable_string,
+    .echo_async_nullable_uint8_list = echo_async_nullable_uint8_list,
+    .echo_async_nullable_object = echo_async_nullable_object,
+    .echo_async_nullable_list = echo_async_nullable_list,
+    .echo_async_nullable_map = echo_async_nullable_map,
+    .echo_async_nullable_enum = echo_async_nullable_enum,
+    .call_flutter_noop = call_flutter_noop,
+    .call_flutter_throw_error = call_flutter_throw_error,
+    .call_flutter_throw_error_from_void = call_flutter_throw_error_from_void,
+    .call_flutter_echo_all_types = call_flutter_echo_all_types,
+    .call_flutter_echo_all_nullable_types =
+        call_flutter_echo_all_nullable_types,
+    .call_flutter_send_multiple_nullable_types =
+        call_flutter_send_multiple_nullable_types,
+    .call_flutter_echo_all_nullable_types_without_recursion =
+        call_flutter_echo_all_nullable_types_without_recursion,
+    .call_flutter_send_multiple_nullable_types_without_recursion =
+        call_flutter_send_multiple_nullable_types_without_recursion,
+    .call_flutter_echo_bool = call_flutter_echo_bool,
+    .call_flutter_echo_int = call_flutter_echo_int,
+    .call_flutter_echo_double = call_flutter_echo_double,
+    .call_flutter_echo_string = call_flutter_echo_string,
+    .call_flutter_echo_uint8_list = call_flutter_echo_uint8_list,
+    .call_flutter_echo_list = call_flutter_echo_list,
+    .call_flutter_echo_map = call_flutter_echo_map,
+    .call_flutter_echo_enum = call_flutter_echo_enum,
+    .call_flutter_echo_nullable_bool = call_flutter_echo_nullable_bool,
+    .call_flutter_echo_nullable_int = call_flutter_echo_nullable_int,
+    .call_flutter_echo_nullable_double = call_flutter_echo_nullable_double,
+    .call_flutter_echo_nullable_string = call_flutter_echo_nullable_string,
+    .call_flutter_echo_nullable_uint8_list =
+        call_flutter_echo_nullable_uint8_list,
+    .call_flutter_echo_nullable_list = call_flutter_echo_nullable_list,
+    .call_flutter_echo_nullable_map = call_flutter_echo_nullable_map,
+    .call_flutter_echo_nullable_enum = call_flutter_echo_nullable_enum,
+    .call_flutter_small_api_echo_string = call_flutter_small_api_echo_string};
+
+static void echo(const gchar* a_string,
+                 CoreTestsPigeonTestHostSmallApiResponseHandle* response_handle,
+                 gpointer user_data) {
+  core_tests_pigeon_test_host_small_api_respond_echo(response_handle, a_string);
+}
+
+static void void_void(
+    CoreTestsPigeonTestHostSmallApiResponseHandle* response_handle,
+    gpointer user_data) {
+  core_tests_pigeon_test_host_small_api_respond_void_void(response_handle);
+}
+
+static CoreTestsPigeonTestHostSmallApiVTable host_small_api_vtable = {
+    .echo = echo, .void_void = void_void};
+
+static void test_plugin_dispose(GObject* object) {
+  TestPlugin* self = TEST_PLUGIN(object);
+
+  g_cancellable_cancel(self->cancellable);
+
```

Comment:
It shouldn't really matter in the context of the test, but since this is also kind of documentation it would be good to demonstrate the call here to clear the handlers (the same ones set in `test_plugin_new`) that a real plugin should have.

Reply:
I added this (hopefully correctly!)